README.md

```diff
@@ -24,14 +21,77 @@ If you have downloaded the repo without using the `--recurse-submodules` argumen
 git submodule update --init --recursive
 ```
 
+## coreMQTT Agent Library Configurations
+
+The MQTT Agent library exposes build configuration macros that are required for building the library.
+However, the MQTT Agent library does not include its own configuration file, but exposes configurable macros and their default values at the top of [core_mqtt_agent.h](source/include/core_mqtt_agent.h) and [core_mqtt_agent_command_functions.h](source/include/core_mqtt_agent_command_functions.h). Also, the library inherits any macros defined in coreMQTT's `core_mqtt_config.h`.
+
+To provide custom values for the configuration values, they must be either:
+* Defined in `core_mqtt_config.h` used by coreMQTT
+**OR**
+* Passed as compile time preprocessor macros
+
+## Building the Library
+
+The [mqttAgentFilePaths.cmake](mqttAgentFilePaths.cmake) file contains the information of all source files and the header include path from this repository. In addition to these files, the MQTT Agent library requires the coreMQTT library, whose file information can be obtained from `mqttFilePaths.cmake` in the [coreMQTT submodule](source/dependency/).
+
+For a CMake example of building the MQTT Agent library with the `mqttAgentFilePaths.cmake` file, refer to the `coverity_analysis` library target in [test/CMakeLists.txt](test/CMakeLists.txt) file.
+
+## Building Unit Tests
+
+### Checkout CMock Submodule
+
+To build unit tests, the submodule dependency of CMock is required. Use the following command to clone the submodule:
+```
+git submodule update --checkout --init --recursive test/unit-test/CMock
+```
+
+### Platform Prerequisites
```

Comment:
Please make this clear that CMake and Ruby are required to build the tests, not the library.

Reply:
Changed